packages/webapp/src/components-v2/EnvironmentDropdown.tsx

```diff
@@ -0,0 +1,220 @@
+                                            to add more
+                                        </>
+                                    )}
+                                </TooltipContent>
+                            )}
+                        </Tooltip>
+                    </DropdownMenuContent>
+                </DropdownMenu>
+                <CreateEnvironmentDialog open={environmentDialogOpen} onOpenChange={setEnvironmentDialogOpen} />
+            </SidebarMenuItem>
+        </SidebarMenu>
+    );
+};
+
+interface CreateEnvironmentDialogProps {
+    open: boolean;
+    onOpenChange: (open: boolean) => void;
+}
+
+const environmentFormSchema = z.object({
+    name: z
+        .string()
+        .regex(/^[a-z0-9_-]+$/, 'Invalid environment name')
+        .max(255)
+});
+
+type EnvironmentForm = z.infer<typeof environmentFormSchema>;
+
+export const CreateEnvironmentDialog: React.FC<CreateEnvironmentDialogProps> = ({ open, onOpenChange }) => {
+    const form = useForm<EnvironmentForm>({
+        resolver: zodResolver(environmentFormSchema),
+        defaultValues: {
+            name: ''
+        }
+    });
+
+    const { toast } = useToast();
+    const navigate = useNavigate();
+    const { mutate: mutateMeta } = useMeta();
+
+    const [loading, setLoading] = useState(false);
+
+    async function onSubmit(data: EnvironmentForm) {
+        setLoading(true);
+
+        const res = await apiPostEnvironment({ name: data.name });
+        if ('error' in res.json) {
+            const err = res.json.error;
+            if (err.code === 'invalid_body') {
+                form.setError('name', { message: 'Invalid environment name' });
+            } else if (['conflict', 'feature_disabled', 'resource_capped'].includes(err.code)) {
+                toast({ title: err.message, variant: 'error' });
+            } else {
+                toast({ title: 'Failed to create environment', variant: 'error' });
+            }
+        } else {
+            navigate(`/${res.json.data.name}`);
+            onOpenChange(false);
+            form.reset();
+            void mutateMeta();
+        }
+
+        setLoading(false);
+    }
+
+    return (
+        <Dialog open={open} onOpenChange={onOpenChange} modal={true}>
+            <DialogContent>
+                <Form {...form}>
+                    <form onSubmit={form.handleSubmit(onSubmit)} className="flex flex-col gap-10">
+                        <DialogHeader>
+                            <DialogTitle>Environment Name</DialogTitle>
+                        </DialogHeader>
+                        <FormField
+                            control={form.control}
+                            name="name"
+                            render={({ field }) => (
+                                <FormItem>
+                                    <FormControl>
+                                        <Input type="text" placeholder="my-environment-name" {...field} />
```

Comment:
this input triggers my password manager, maybe it's missing autocomplete=off or something